### PR TITLE
Route53 private zone support

### PR DESCRIPTION
**Pull request by hummus**
support for `HostedZone.Config.PrivateZone=True`

### CODE REVIEW

File: moto/emr/__init__.py [hunk 1]
```diff
@@ -6,7 +6,4 @@
 
 
 def mock_emr(func=None):
-    if func:
-        return MockAWS(emr_backends)(func)
-    else:
-        return MockAWS(emr_backends)
+    return MockAWS(emr_backends)
```
Comment (spulec) [inline moto/emr/__init__.py]: Why this change?

**Comment by spulec** [comment]
Thanks for this. One question, but otherwise happy to merge

File: moto/emr/__init__.py (hunk 1 above)
Comment (hummus) [inline moto/emr/__init__.py]: Oh my. Sorry, this disabled the emr testing while I somehow thought it had just fixed my issue where `test_emr.test_create_job_flow_in_multiple_regions` would not pass. 

passes in CI now though I can't get it pass locally....

**Comment by spulec** [comment]
Great, thanks!
